Tags review: which tags are in which contrasts?

ferponcem opened the pull request:
This PR is a continuation of the efforts started in #73, just that now we're looking at every tag and listing the contrasts annotated with it. From there, we can suggest to maintain/remove/annotate more contrasts with it.

This initial commit summarizes the discussions of 04/Dec and 11/Dec/2024, with more to come.
Staring with the tags:
- [naturalistic_scenes ](https://www.cognitiveatlas.org/concept/id/trm_EibcrcPYJ86gd/): added it to flora and fruits visual stimuli, without the capital letters
- removed [attentional_focusing](https://www.cognitiveatlas.org/concept/id/trm_4a3fd79d0b5ef/) in favour of [selective_attention](https://www.cognitiveatlas.org/concept/id/trm_4a3fd79d0b043/). to me, this two refer to the same thing, but `selective_attention` appears more consistently through our contrasts
- removed [auditory_imagery](https://www.cognitiveatlas.org/concept/id/trm_4a3fd79d09a6c/) as we don't have any task that ask subjects to imagine sounds, please correct me if I'm wrong
- [auditory_attention](https://www.cognitiveatlas.org/concept/id/trm_4a3fd79d09a10/) is left only in control conditions, otherwise it's redundant 
- [auditory_recognition](https://www.cognitiveatlas.org/concept/id/trm_55ef687f48a9d/) is now in all audio-stimuli when there are several categories
- [body_maintenance](https://www.cognitiveatlas.org/concept/id/trm_557b477b4a15f/) was added to the visual body stimuli from the `MultiModal` task, along with `face_maintenance` and `object_maintenance`, as this is a 1-back working mem task
- added [color_perception](https://www.cognitiveatlas.org/concept/id/trm_4a3fd79d09d93/) to other tasks where color places a role (say which color happend more, etc)
- added [decision_making](https://www.cognitiveatlas.org/concept/id/trm_4a3fd79d0a038/) to tasks where subjects have to choose between 2 or more explicit options

I'll keep updating with the following discussions, please let me know what you think (:

bthirion reviewed (commented):
LGTM so far

ferponcem commented:
Some comments on the latest pass:
- In [commit 4f618d3](https://github.com/individual-brain-charting/public_analysis_code/pull/81/commits/4f618d3eb0bd5d32666a64bc0bb725c4ed421e7a)
    - added `emotional_face_recognition` to some contrasts in ArchiEmotional and FacesAomic
    - added `episodic_memory` to some of the contrasts from MTTWE and NS: to all the events + sn_average_reference
- In [commit ae49b1c](https://github.com/individual-brain-charting/public_analysis_code/pull/81/commits/ae49b1c7388c760a7f621dbf7a61acd14e7cd4f0)
    - added `feature_comparison` to some contrasts in MVIS, DotPatterns and WMAomic. I also added it to HarririAomic but I'm having second thoughts on adding it to the *shape* contrast
    - added `facial_expression` to HarririAomic and EmoReco
- In [commit f8927d8](https://github.com/individual-brain-charting/public_analysis_code/pull/81/commits/f8927d841cbbd1043c38721b81934d5493b33d9c)
    - added food_cue_reactivity to fruits stimuli: Visu and MultiModal tasks

I am having trouble with `far-peripheral_vision` because in the meeting we discussed we should probably just leave it as *peripheral_vision* but looking at it the tag `mid-peripheral_vision` also exists, to distinguish visual representation in the mid and far periphery of the visual field.
In the cognitive atlas, the [Retinotopic task description](https://www.cognitiveatlas.org/task/id/trm_558c4d3105abf/) has no associated concepts so I guess we're free of defining it. I'd propose to leave it as *far_peripheral_vision* removing the "-" for an "_" for consistency but keeping both.

bthirion commented:
Regarding `far-peripheral` and `mid-peripheral`, I agree with your suggestion to move to `far_peripheral` and `mid_peripheral`

ferponcem commented:
Thanks for the comments ! Moving along:
- [commit e8ea025](https://github.com/individual-brain-charting/public_analysis_code/pull/81/commits/e8ea0259d0696b2efaeae9cb06589d9a56c769ef):
    - modified to *mid_* and *far_peripheral_vision*
    - added `future_time` to 2 contrasts in OptimismBias where the effect is also present, in my opinion: *future-past* and *future_positive-negative*
- [commit 4cd22c6](https://github.com/individual-brain-charting/public_analysis_code/pull/81/commits/4cd22c6c8a9bc7cf517316a559bf77fd21c1fa06):
    - for `future_time`, I added this to some instances of OptimismBias. I was afraid it would be repetitive of  `episodic_future_thinking`, but the former is more general about the future, and the latter is really focused on self's projection, so in this case I am hoping they're complementary.
    - in the cases of `gender_discrimination` and `gender_perception`, I am struggling to make the distinction clear (these two are not part of the cognitive atlas but I am keeping them anyway). In my understanding, **gender_discrimination** is the active decision of "okay, this person is a woman", meaning it's an explicit output, and on the other hand, **gender_perception** is not necessarily conscious, you just see a face and internalize it but you're not required to say whether it's a man or a woman. following this logic, I tried to put them together in all of the instances where we requested participants to indicate the gender of a person, I think they're complementary. The only exception is the *FacesAomic* task, where they are asked to remember the face and the expression, so I am thinking the gender is not explicitly reported but makes part of it.
- commit [73ccb49](https://github.com/individual-brain-charting/public_analysis_code/pull/81/commits/73ccb494ac5d4f4dd4fa0e89a6776bb933f3bed4):
    - ` auditory_attention`: I was using this to only model silence, but now I think this is misleading, and instead it can be complementary to `auditory_perception` in some cases. I added it to the contrasts for *Audi*, since participants are asked to pay attention and at the end, report if they heard a sound or not. I also added it to the `LPPLocalizer` and to the audio- contrasts on ArchiStandard. On the other hand, I didn't add it to *Audio* or *MultiModal* tasks, since those are working memory tasks (you're asked to reply in the immediate next trial if the stimuli is repeated), so I think the interesting effect is the working memory one.

ferponcem commented:
Next, I wanted to revise the `shape_perception` tag, and I figured this was present mostly on inhibition tasks, but it should be used in other contrasts in my opinion:

In commit [5cfa935](https://github.com/individual-brain-charting/public_analysis_code/pull/81/commits/5cfa9356bbcbe4e87463b6739b3257ddd333a510), I did the following:
- I added `shape_perception` to some contrasts were we are also using `shape_recognition`. In my understanding, perception is about attending to shapes and recognition is more about identifying a shape that was seen previously. In this sense, most of the time that shapes are involved, shape_perception should be there. I added it for tasks like HcpRelational. For tasks like HcpEMotion, HcpSocialand Enumeration I used to replace shape_recognition, since it's not really a task related to the shape (you're not asked to identify the shape) but to something else.
- I did a bunch of changes on the SelectiveStopSignal and StopSignal tasks, but the most important is that I added shape_perception, again based on my premise of attending to shapes involves it.
- For the DotPatterns task, I added numerosity and removed shape_recognition, because the task is based on recognizing the spatial distribution of shapes, not the shapes themselves (they're always circles). That's why I added numerosity to some of them, because there are some different amount of circles every trial. 
- For the StopNoGo task, I removed shape_recognition as well, since the task is not about the shape (it's always an arrow), but about the color of the arrow. 

Let me know what you think !

bthirion commented:
> Next, I wanted to revise the `shape_perception` tag, and I figured this was present mostly on inhibition tasks, but it should be used in other contrasts in my opinion:
> 
> In commit [5cfa935](https://github.com/individual-brain-charting/public_analysis_code/pull/81/commits/5cfa9356bbcbe4e87463b6739b3257ddd333a510), I did the following:
> 
>     * I added `shape_perception` to some contrasts were we are also using `shape_recognition`. In my understanding, perception is about attending to shapes and recognition is more about identifying a shape that was seen previously. In this sense, most of the time that shapes are involved, shape_perception should be there. I added it for tasks like HcpRelational. For tasks like HcpEMotion, HcpSocialand Enumeration I used to replace shape_recognition, since it's not really a task related to the shape (you're not asked to identify the shape) but to something else.
> 
>     * I did a bunch of changes on the SelectiveStopSignal and StopSignal tasks, but the most important is that I added shape_perception, again based on my premise of attending to shapes involves it.
> 
>     * For the DotPatterns task, I added numerosity and removed shape_recognition, because the task is based on recognizing the spatial distribution of shapes, not the shapes themselves (they're always circles). That's why I added numerosity to some of them, because there are some different amount of circles every trial.
> 
>     * For the StopNoGo task, I removed shape_recognition as well, since the task is not about the shape (it's always an arrow), but about the color of the arrow.
> 
> 
> Let me know what you think !

Thx ! I agree with what you propose.
Best,

bthirion reviewed (approved):
LGTM. I think we can merge this PR before doing more changes.
